glean-core/tests/string_list.rs

```diff
@@ -23,17 +26,142 @@ fn list_can_store_multiple_items() {
+    }
+
+    {
+        let glean = Glean::new(&tmpname, GLOBAL_APPLICATION_ID, true).unwrap();
+
+        let snapshot = StorageManager
+            .snapshot_as_json(glean.storage(), "store1", true)
+            .unwrap();
+        assert_eq!(
+            json!({"string_list": {"telemetry.test.string_list_metric": ["test_string_1", "test_string_2"]}}),
+            snapshot
+        );
+    }
+}
+
+#[test]
+fn set_properly_sets_the_value_in_all_stores() {
+    let (glean, _t) = new_glean();
+    let store_names: Vec<String> = vec!["store1".into(), "store2".into()];
+
+    let metric = StringListMetric::new(CommonMetricData {
+        name: "string_list_metric".into(),
+        category: "telemetry.test".into(),
+        send_in_pings: store_names.clone(),
+        disabled: false,
+        lifetime: Lifetime::Ping,
+    });
+
+    metric.set(&glean, vec!["test_string_1".into(), "test_string_2".into()]);
+
+    for store_name in store_names {
+        let snapshot = StorageManager
+            .snapshot_as_json(glean.storage(), &store_name, true)
+            .unwrap();
+
+        assert_eq!(
+            json!({"string_list": {"telemetry.test.string_list_metric": ["test_string_1", "test_string_2"]}}),
+            snapshot
+        );
+    }
+}
+
+#[test]
+fn long_string_values_are_truncated() {
+    let (glean, _t) = new_glean();
+
+    let metric = StringListMetric::new(CommonMetricData {
+        name: "string_list_metric".into(),
+        category: "telemetry.test".into(),
+        send_in_pings: vec!["store1".into()],
+        disabled: false,
+        lifetime: Lifetime::Ping,
+    });
+
+    let test_string = "0123456789".repeat(20);
+    metric.add(&glean, test_string.clone());
+
+    // Ensure the string was truncated to the proper length.
+    assert_eq!(
+        format!("[\"{}\"]", test_string[..50].to_string()),
+        metric.test_get_value_as_string(&glean, "store1").unwrap()
+    );
+
+    // Ensure the error has been recorded.
+    assert_eq!(
+        Ok(1),
+        test_get_num_recorded_errors(&glean, metric.meta(), ErrorType::InvalidValue, None)
+    );
+}
+
+#[test]
+fn disabled_string_lists_dont_record() {
+    let (glean, _t) = new_glean();
+
+    let metric = StringListMetric::new(CommonMetricData {
+        name: "string_list_metric".into(),
+        category: "telemetry.test".into(),
+        send_in_pings: vec!["store1".into()],
+        disabled: true,
+        lifetime: Lifetime::Ping,
+    });
+
+    metric.add(&glean, "test_string".repeat(20));
+
+    // Ensure the string was not added.
+    assert_eq!(None, metric.test_get_value_as_string(&glean, "store1"));
+
+    metric.set(&glean, vec!["test_string_2".repeat(20)]);
+
+    // Ensure the stringlist was not set.
+    assert_eq!(None, metric.test_get_value_as_string(&glean, "store1"));
+
+    // Ensure no error was recorded.
+    assert!(
+        test_get_num_recorded_errors(&glean, metric.meta(), ErrorType::InvalidValue, None).is_err()
+    );
```

Comment:
This lacks a test that we limit the number of elements in a list.
We should check that

* We can add N items
* On adding the N+1 item the list stays unchanged (and I think we log an error?)